packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -872,9 +850,9 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
 
   final Set<int> _acceptedActivePointers = <int>{};
 
-  bool _hasSufficientGlobalDistanceToAccept(PointerDeviceKind pointerDeviceKind, double? deviceTouchSlop) {
-    return _globalDistanceMoved.abs() > computePanSlop(pointerDeviceKind, gestureSettings);
-  }
+  Offset _getDeltaForDetails(Offset delta);
```

Comment:
This should be public, otherwise this abstract class can't be used outside of this file, you might as well make the abstract class private

Reply:
Also these seems like a vary limited API, I would probably suggest toward making _checkDrag more customizable if this class meant to be general purposes.

If the `TapAndDragGestureRecognizer` meant to be private, then I think this is fine.

Reply:
I'm a little on the fence about this since `DragGestureRecognizer` has these same methods private and I would prefer to keep them private unless there is an immediate use case for it. 

My goal for this PR is mostly to fix the scrolling issue in `TextField`.
I can always expose these methods in a later PR if there is a need for it.

I think keeping `BaseTapAndDragGestureRecognizer` public is fine, since it doesn't make it any more or less extensible than `TapAndDragGestureRecognizer` initially was. What do you think?

Reply:
Anyone would try to extends it outside this file would crash. The current implementation will create a foot gun

I think there are four ways to do this

1. Make these methods public
2. Make this class private
3. Turns constructor into private constructor (so no one can override)
4. give these method default implementations

Reply:
Thank you for further clarifying the problem that this causes and your suggestions!  Maybe another option is also to make `BaseTapAndDragGestureRecognizer` a `sealed` class.

```dart
class TapAndDragTestExtension extends BaseTapAndDragGestureRecognizer {

}
```

Trying this outside of the file results in the following error: `The class 'BaseTapAndDragGestureRecognizer' can't be extended, implemented, or mixed in outside of its library because it's a sealed class.dart[invalid_use_of_type_outside_library](https://dart.dev/diagnostics/invalid_use_of_type_outside_library)`.

This should avoid the foot gun and also allow us to keep the documentation on `BaseTapAndDragGestureRecognizer`.